src/main/java/org/elasticsearch/script/CompiledScript.java

```diff
@@ -19,21 +19,34 @@
 
 package org.elasticsearch.script;
 
+import static org.elasticsearch.script.ScriptService.ScriptType;
+
 /**
  *
  */
 public class CompiledScript {
 
-    private final String type;
-
+    private final String lang;
+    private final String name;
+    private final ScriptType type;
```

Comment:
are name and scriptType needed here? If I understood correctly type becomes lang (good!) and we also have the compiled script itself right? anything more we need?